js/glpi_install.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+(() => {
+    function message(message_list_element, text) {
+        const alert = document.createElement('p');
+        alert.innerHTML = text;
+        message_list_element.appendChild(alert);
+    }
+
+    async function start_database_install(dom_element, progress_key)
+    {
+        if (!dom_element) {
+            throw new Error('No DOM element provided to start database install.');
+        }
+        const messages_container = document.getElementById('glpi_install_messages_container');
+        const success_element = document.getElementById('glpi_install_success');
+
+        const message_list_element = document.createElement('div');
+
+        success_element.querySelector('button').setAttribute('disabled', 'disabled');
+
+        const create_progress_bar = window.create_progress_bar;
+
+        if (typeof create_progress_bar === 'undefined') {
+            throw new Error('Function "create_progress_bar" is not defined. Did you load the associated JS file correctly?');
+        }
+
+        const progress = create_progress_bar({
+            key: progress_key,
+            container: messages_container,
+            error_callback: (error) => message(message_list_element, `Progress error:\n${error}`),
```

Comment:
We should not display technical messages to the end-user as long as it will not help him to fix it by itself.
If someone ask for support, we would tell hims to check his logs in order to give us the corresponding stack trace.

Also, all messages should be translated (with the `__()` function). I will open a PR to fix the locales loading on the install page.

```suggestion
            error_callback: (error) => message(message_list_element, __('An unexpected error has occurred.')),
```